test/visual/freedraw.js

```diff
@@ -2056,9 +2015,43 @@
     }
   ].map(p => new fabric.Point(p));;
 
-  var tests = [];
+  
 
-  var freeDrawingTestDefaults = {
+QUnit.module('Free Drawing', hooks => {
+  const visualTester = visualTestLoop(QUnit);
+  let objectCachingDefault;
+  const BROWSER_SHADOW_BLUR = {
+    firefox: 0.9,
+    chrome: 1.5,
+    edge: 1.75,
+    node: 1
+  };
+  function getBrowserAgent() {
+    return ['chrome', 'firefox', 'edge'].find(name => navigator.userAgent.toLowerCase().indexOf(name) > -1);
+  }
+  hooks.before(() => {
+    objectCachingDefault = fabric.Object.prototype.objectCaching;
+    if (fabric.isLikelyNode) {
+      fabric.config.configure({
+        browserShadowBlurConstant: BROWSER_SHADOW_BLUR[process.env.launcher?.toLowerCase() || 'node']
+      });
+    }
+    else {
+      fabric.config.configure({ browserShadowBlurConstant: BROWSER_SHADOW_BLUR[getBrowserAgent()] });
+    }
+    fabric.config.configure({
+      enableGLFiltering: false
+    });
+    fabric.Object.prototype.objectCaching = true;
+  });
+  hooks.after(() => {
+    // fabric.config.restoreDefaults();
```

Comment:
This is really bad.
Other suites depend on some configuration made in this test that leaks.